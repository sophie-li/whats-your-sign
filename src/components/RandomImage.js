import React, { Component } from 'react';

class RandomImage extends Component {
  constructor(props) {
    super(props);
    this.state = { imageID: 1 };
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.imageID !== nextState.imageID) {
      return true;
    }
    return false;
  }

  updateImageID = (e) => {
    const imageID = Math.floor(Math.random() * 50) + 1; // Random number between 1 to 50
    this.setState(() => {
      return { imageID: imageID };
    });
  };

  render() {
    const { imageID } = this.state;
    const url = process.env.PUBLIC_URL + `/images/magic/magic-${imageID}.png`;
    return (
      <img
        src={url}
        className='magic-img'
        onClick={this.updateImageID}
        alt='random magic'
      />
    );
  }
}

export default RandomImage;
