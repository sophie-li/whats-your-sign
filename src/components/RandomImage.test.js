import { shallow } from 'enzyme';
import React from 'react';
import RandomImage from './RandomImage';

it('expect to render RandomImage component', () => {
  expect(shallow(<RandomImage />)).toMatchSnapshot();
});

it('correctly changes imageID', () => {
  const wrapper = shallow(<RandomImage />);

  wrapper.find('[id="magic-img"]').simulate('click');
  let imageID = wrapper.state().imageID;
  expect(imageID).toBeGreaterThanOrEqual(1);
  expect(imageID).toBeLessThan(51);

  wrapper.find('[id="magic-img"]').simulate('click');
  imageID = wrapper.state().imageID;
  expect(imageID).toBeGreaterThanOrEqual(1);
  expect(imageID).toBeLessThan(51);
});
