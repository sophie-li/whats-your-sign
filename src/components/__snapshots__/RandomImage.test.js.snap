// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render RandomImage component 1`] = `
<img
  alt="random magic"
  className="magic-img"
  id="magic-img"
  onClick={[Function]}
  src="/images/magic/magic-1.png"
/>
`;
